plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -329,9 +336,26 @@ func (r *ReferenceManager) ensureShootReferences(shoot *garden.Shoot) error {
 		return err
 	}
 
+	if isSpecifiedCustomAuditPolicy(shoot.Spec.Kubernetes.KubeAPIServer) {
+		if _, err := r.configMapLister.ConfigMaps(shoot.Namespace).Get(shoot.Spec.Kubernetes.KubeAPIServer.AuditConfig.AuditPolicy.Name); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+func isSpecifiedCustomAuditPolicy(apiServerConfig *garden.KubeAPIServerConfig) bool {
+	if apiServerConfig != nil && apiServerConfig.AuditConfig != nil {
+		if apiServerConfig.AuditConfig.AuditPolicy != nil {
+			if len(apiServerConfig.AuditConfig.AuditPolicy.Name) != 0 {
```

Comment:
Please merge the `if` conditions and combine them with `&&` (reduces too deeply nested expressions).